templates/lib/yarp/serialize.rb.erb

```diff
@@ -89,8 +89,13 @@ module YARP
         unless constant
           offset = constant_pool_offset + index * 8
 
-          start = serialized.unpack1("L", offset: offset)
-          length = serialized.unpack1("L", offset: offset + 4)
+          if RUBY_VERSION >= "3.1"
```

Comment:
This check is slow (so it slows downs the fast path on >= 3.1).

Reply:
Agreed. I would prefer a polyfill to a version check here.